app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt

```diff
@@ -69,21 +77,34 @@ class LeanplumMetricsService(private val application: Application) : MetricsServ
         Parser.parseVariables(application)
 
         LeanplumActivityHelper.enableLifecycleCallbacks(application)
-
-        val installedApps = MozillaProductDetector.getInstalledMozillaProducts(application)
-
-        Leanplum.start(application, hashMapOf(
-            "default_browser" to MozillaProductDetector.getMozillaBrowserDefault(application).orEmpty(),
-            "fennec_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.FIREFOX.productName),
-            "focus_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.FOCUS.productName),
-            "klar_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.KLAR.productName),
-            "fxa_signed_in" to application.settings().fxaSignedIn,
-            "fxa_has_synced_items" to application.settings().fxaHasSyncedItems,
-            "search_widget_installed" to application.settings().searchWidgetInstalled
-        ))
+        Leanplum.start(application)
+
+        lpmsSetUserAttributeJob = lpmsSetUserAttributeCoroutineScope.launch {
+            val installedApps = MozillaProductDetector.getInstalledMozillaProducts(application)
+            Leanplum.setUserAttributes(
```

Comment:
Looking at `Leanplum`, I don't see anything providing thread-safety or visibility guarantees. Why are we confident that it is safe to set the user attributes later on a different thread? To me it looks like we need to switch back to main before calling `setUserAttributes`.

If we fixed that we would still make the assumption that it's OK to start without those attributes and set them later. I don't know enough about Leanplum to be confident this works as expected.

Reply:
> Looking at `Leanplum`, I don't see anything providing thread-safety or visibility guarantees. Why are we confident that it is safe to set the user attributes later on a different thread? To me it looks like we need to switch back to main before calling `setUserAttributes`.

`setUserAttributesInternal`, what `setUserAttributes` ultimately uses to do its work, calls `RequestOld.post()` which gets an instance of a `RequestFactory` and uses that factory's `createRequest` which calls the constructor of `RequestOld` which uses code that seems to rely on the fact that it is on the main thread.

However, the code that relies on the fact that it is on the main thread is instantiating that must be instantiated on the main thread. That singleton, however, is *always* instantiated before this call because `Leanplum.start()` instantiates it as its first order of business. 

> 
> If we fixed that we would still make the assumption that it's OK to start without those attributes and set them later. I don't know enough about Leanplum to be confident this works as expected.

After a very detailed discussion with @csadilek IRL, we decided that that risk of setting user attributes asynchronously is *too* great when compared with any potential performance improvement. @csadilek and I came up with an alternate approach and we are going to consider that. 

I am closing this PR as a result.